### PR TITLE
build: use Go 1.23 to build the project

### PR DESCRIPTION
**Pull request by joschi**
https://tip.golang.org/doc/go1.23

### CODE REVIEW

File: database/redshift/redshift_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 var (
 	opts  = dktest.Options{PortRequired: true, ReadyFunc: isReady}
 	specs = []dktesting.ContainerSpec{
-		{ImageName: "postgres:8", Options: opts},
+		{ImageName: "joschi/postgres8:8", Options: opts},
```
Comment (shriacquia) [inline database/redshift/redshift_test.go:34]: Why is it needed to pull this image from a non-standard registry location?

Comment (joschi) [reply database/redshift/redshift_test.go:34]: @shriacquia See https://github.com/golang-migrate/migrate/pull/1128.

**Comment by dhui** [comment]
@joschi this should probably be the next PR that's merge. After that, the other PRs can be rebased off this/master and merged

**Comment by joschi** [comment]
@dhui I've rebased this PR onto the latest default branch.

**Review by dhui — APPROVED** [review]

**Comment by Sora233** [comment]
Sadly, This made me impossible to use this version. 
As The latest go version is 1.23，it is reasonable to support last two major version,  >= go1.21.
This project is not only an cli, but also a library.

**Pull request merged**